### ISSUE TITLE
Deprecate immutable object

### ISSUE THREAD

**Issue by jbergfi** [immutables/immutables#480]
I have an immutable value object I would want to deprecate but I'm only able to deprecate the Abstract class, the generated class is not marked deprecated.

I've tried:

```
/**
 * @deprecated
 */
@Immutable
@Deprecated
static abstract class AbstractMyValue {
}
```

but the generated class is not deprecated:

```
/**
 * Immutable implementation of ...
 */
@SuppressWarnings({"all"})
@ParametersAreNonnullByDefault
@Generated({"Immutables.generator", "AbstractMyValue"})
@Immutable
public final class MyValue extends AbstractMyValue {
}
```

Is it somehow possible to get the generated class marked deprecated?

**Comment by elucash** [comment]
Thank you for the feature request! We'll expand the deprecation support to types

**Comment by bartosz-bilicki** [comment]
As workaround, maybe Value.Style.additionalJsonAnnotation can be used to pass @Deprecated annotation to generated Immutable* class?

**Comment by elucash** [comment]
It may work but cannot say it will, there are places where we filter out `java.lang.*` annotations.

**Comment by elucash** [comment]
the change is in master, will be released in 2.3.9 Thank you!

**Comment by jbergfi** [comment]
Thank you!

**Comment by elucash** [comment]
released in 2.3.9

**Comment by mgreene** [comment]
We've just upgraded to the `2.5.6` to pick up this change and it doesn't seem to be propagating the `@deprecated` annotation when used at the class level like the OP had outlined. Method level seems to work though.

I guess my assumptions is that it would deprecate the builder at the very least.
